src/VisualStudio/ExternalAccess/FSharp/Internal/SignatureHelp/FSharpSignatureHelpProvider.cs

```diff
@@ -12,32 +10,58 @@
 using Microsoft.CodeAnalysis.ExternalAccess.FSharp.SignatureHelp;
 using Microsoft.CodeAnalysis.Host.Mef;
 using Microsoft.CodeAnalysis.SignatureHelp;
+using System.Collections.Immutable;
 
 namespace Microsoft.CodeAnalysis.ExternalAccess.FSharp.Internal.SignatureHelp;
 
 [Shared]
 [ExportSignatureHelpProvider(nameof(FSharpSignatureHelpProvider), LanguageNames.FSharp)]
 internal class FSharpSignatureHelpProvider : ISignatureHelpProvider
 {
-    private readonly IFSharpSignatureHelpProvider _provider;
+#pragma warning disable CS0618 // Type or member is obsolete
+    private readonly IFSharpSignatureHelpProvider? _legacyProvider;
+#pragma warning restore CS0618 // Type or member is obsolete
+    private readonly AbstractFSharpSignatureHelpProvider? _newProvider;
 
     [ImportingConstructor]
     [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-    public FSharpSignatureHelpProvider(IFSharpSignatureHelpProvider provider)
+    public FSharpSignatureHelpProvider(
+        [Import(AllowDefault = true)] IFSharpSignatureHelpProvider? legacyProvider,
+        [Import(AllowDefault = true)] AbstractFSharpSignatureHelpProvider? newProvider)
     {
-        _provider = provider;
+        _legacyProvider = legacyProvider;
+        _newProvider = newProvider;
     }
 
-    public async Task<SignatureHelpItems> GetItemsAsync(Document document, int position, SignatureHelpTriggerInfo triggerInfo, MemberDisplayOptions options, CancellationToken cancellationToken)
+    /// <summary>
+    /// Hard coded from https://github.com/dotnet/fsharp/blob/main/vsintegration/src/FSharp.Editor/Completion/SignatureHelp.fs#L708
+    /// </summary>
+    public ImmutableArray<char> TriggerCharacters => _newProvider is not null ? _newProvider.TriggerCharacters : ['(', '<', ',', ' '];
```

Comment:
this was the easiest way I could come up with to avoid a dual insertion.

once we have packages inserted I'll submit a PR to F# to switch to the new properties, then we can delete the hardcoded characters.

Reply:
yup.  this makes sense to me.